native/src/main/java/io/ballerina/stdlib/etl/utils/CommonUtils.java

```diff
@@ -0,0 +1,254 @@
+        ArrayType arrayType = TypeCreator.createArrayType(TypeUtils.getReferredType(type.getDescribingType()));
+        return (BArray) JsonUtils.convertJSON(source, TypeCreator.createArrayType(arrayType));
+    }
+
+    public static Object convertJSONToRecord(Object source, BTypedesc type) {
+        Type describingType = TypeUtils.getReferredType(type.getDescribingType());
+        return JsonUtils.convertJSON(source, describingType);
+    }
+
+    public static BString[] getFields(BTypedesc type) {
+        Type describingType = TypeUtils.getReferredType(type.getDescribingType());
+        if (!(describingType.getTag() == TypeTags.RECORD_TYPE_TAG)) {
+            ErrorUtils.createInvalidReturnTypeError();
+        }
+        StructureType structType = (StructureType) describingType;
+        Map<String, Field> fields = structType.getFields();
+        BString[] fieldNames = new BString[fields.size()];
+        int i = 0;
+        for (Map.Entry<String, Field> entry : fields.entrySet()) {
+            fieldNames[i++] = StringUtils.fromString(entry.getKey());
+        }
+        return fieldNames;
+    }
+
+    public static boolean isFieldExist(BArray dataset, BString fieldName) {
+        Type describingType = TypeUtils.getReferredType(dataset.getElementType());
+        if (!(describingType.getTag() == TypeTags.RECORD_TYPE_TAG)) {
+            ErrorUtils.createInvalidReturnTypeError();
+        }
+        StructureType structType = (StructureType) describingType;
+        Map<String, Field> fields = structType.getFields();
+        return fields.containsKey(fieldName.getValue());
+    }
+
+    public static Type getFieldType(BTypedesc type, BString fieldName) {
+        Type describingType = TypeUtils.getReferredType(type.getDescribingType());
+        if (!(describingType.getTag() == TypeTags.RECORD_TYPE_TAG)) {
+            ErrorUtils.createInvalidReturnTypeError();
+        }
+        StructureType structType = (StructureType) describingType;
+        Map<String, Field> fields = structType.getFields();
+        return fields.get(fieldName.getValue()).getFieldType();
+    }
+
+    public static boolean isStringType(Type type) {
+        if (type.getTag() == TypeTags.UNION_TAG) {
+            List<Type> memberTypes = ((UnionType) type).getMemberTypes();
+            for (Type memberType : memberTypes) {
+                if (memberType.getTag() == TypeTags.STRING_TAG) {
+                    return true;
+                }
+            }
+            return false;
+        }
+        return type.getTag() == TypeTags.STRING_TAG;
+    }
+
+    public static boolean isNumericType(Type type) {
+        if (type.getTag() == TypeTags.UNION_TAG) {
+            List<Type> memberTypes = ((UnionType) type).getMemberTypes();
+            for (Type memberType : memberTypes) {
+                if (memberType.getTag() == TypeTags.INT_TAG || memberType.getTag() == TypeTags.FLOAT_TAG) {
+                    return true;
+                }
+            }
```

Comment:
Since the union type check is duplicated, I guess we can create a single method to check a given type is a member of the union type and use it everywhere.